### ISSUE TITLE
[Suspend] Appeal suspensions functionality

### ISSUE THREAD

**Issue by davwheat** [flarum/core#3010]
## Feature Request

**Is your feature request related to a problem? Please describe.**
Allow suspended user to appeal their suspensions. This should ideally show similarly to a report to admins.

**Describe the solution you'd like**
Allow suspended users to appeal suspensions a set number of times if they choose.

This could take inspiration from existing functionality in Flarum's reports extension, allowing a user to explain why they should be unsuspended.

Ideally, we'd have an appeal as a discussion that can be replied to multiple times. A decision on the appeal could show as an event post, and integration with flarum/lock could be used to lock the post when a decision has been reached.

**Justify why this feature belongs in Flarum's core, rather than in a third-party extension**
This is a tricky one. Integration into suspend for appeals would be best in the core extension because it's a fairly standard feature that comes alongside bans in other software, but it's entirely possible for it to be a community extension too. This is up to us to decide.

**Describe alternatives you've considered**
We could implement this idea into or steal some code from Byobu if we wanted, as it handles private discussions and such well.

**Comment by luceos** [comment]
I am leaning heavily towards leaving this to the community and not making it a responsibility of core. The more dependent extensions a feature has the easier we feel that functionality needs to be bundled and bundled extensions merged etc. We should warrant that over-complexity isn't part of our core code.

**Comment by askvortsov1** [comment]
> I am leaning heavily towards leaving this to the community and not making it a responsibility of core. The more dependent extensions a feature has the easier we feel that functionality needs to be bundled and bundled extensions merged etc. We should warrant that over-complexity isn't part of our core code.

Agreed. We should make sure suspend is extensible enough to support that, but there's a LOT of different use cases and needs for an appeals system, and it'd be hard for us to cover all of them.
